File: ios/LaneShadow/Views/Molecules/LSTagPill.swift
import LaneShadowTheme
import SwiftUI

struct LSTagPillResolvedStyle {
    let backgroundToken: String
    let borderToken: String
    let iconColor: IconContentColor
    let backgroundColor: Color
    let borderColor: Color
}

public struct LSTagPill: View {
    @Environment(\.theme) private var theme

    let icon: IconName?
    let labelText: String
    let size: PillSize

    var resolvedStyle: LSTagPillResolvedStyle {
        LSTagPillResolvedStyle(
            backgroundToken: "color.surface.glass",
            borderToken: "color.border.default",
            iconColor: .signal,
            backgroundColor: LaneShadowTheme.color.surface.glass,
            borderColor: LaneShadowTheme.color.border.default
        )
    }

    public init(
        icon: IconName? = nil,
        label: String,
        size: PillSize = .sm
    ) {
        self.icon = icon
        labelText = label
        self.size = size
    }

    public var body: some View {
        let style = resolvedStyle

        LSPill(size: size) {
            HStack(spacing: theme.space.xs) {
                if let icon {
                    LSIcon(name: icon, size: .xs, color: style.iconColor)
                }

                LSText(labelText, variant: .label.sm, color: .secondary)
            }
        }
        .background(
            RoundedRectangle(cornerRadius: theme.radius.full, style: .continuous)
                .fill(style.backgroundColor)
        )
        .overlay(
            RoundedRectangle(cornerRadius: theme.radius.full, style: .continuous)
                .stroke(style.borderColor, lineWidth: theme.borderWidth.hairline)
        )
    }
}

File: android/app/src/main/java/com/laneshadow/ui/molecules/LSTagPill.kt
package com.laneshadow.ui.molecules

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.laneshadow.theme.LocalLaneShadowTheme
import com.laneshadow.theme.generated.LaneShadowTheme.IconName
import com.laneshadow.ui.atoms.IconSize
import com.laneshadow.ui.atoms.LSIcon
import com.laneshadow.ui.atoms.LSPill
import com.laneshadow.ui.atoms.LSText
import com.laneshadow.ui.atoms.PillSize

@Composable
fun LSTagPill(
    label: String,
    icon: IconName? = IconName.Pin,
    accent: AccentColor = AccentColor.Muted,
    size: PillSize = PillSize.Sm,
    modifier: Modifier = Modifier,
) {
    val theme = LocalLaneShadowTheme.current
    val style = resolveTagPillStyle(accent)

    LSPill(
        size = size,
        modifier = modifier.tagPillSurface(style = style, cornerRadius = theme.radius.full),
    ) {
        if (icon != null) {
            LSIcon(
                name = icon,
                size = IconSize.Xs,
                color = style.iconColor,
            )
        }
        LSText(
            text = label,
            variant = style.labelVariant,
            color = style.labelColor,
        )
    }
}

private fun Modifier.tagPillSurface(
    style: TagPillStyle,
    cornerRadius: androidx.compose.ui.unit.Dp,
): Modifier {
    val shape = RoundedCornerShape(cornerRadius)
    return this
        .background(style.backgroundColor, shape)
        .border(1.dp, style.borderColor, shape)
}
